cache/src/main/java/io/envoyproxy/controlplane/cache/V3Snapshot.java

```diff
@@ -0,0 +1,228 @@
+package io.envoyproxy.controlplane.cache;
+
+import static io.envoyproxy.controlplane.cache.Resources.TYPE_URLS_TO_RESOURCE_TYPE;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
+import com.google.protobuf.Message;
+import io.envoyproxy.controlplane.cache.Resources.ResourceType;
+import io.envoyproxy.envoy.config.cluster.v3.Cluster;
+import io.envoyproxy.envoy.config.endpoint.v3.ClusterLoadAssignment;
+import io.envoyproxy.envoy.config.listener.v3.Listener;
+import io.envoyproxy.envoy.config.route.v3.RouteConfiguration;
+import io.envoyproxy.envoy.extensions.transport_sockets.tls.v3.Secret;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * {@code Snapshot} is a data class that contains an internally consistent snapshot of v3 xDS
+ * resources. Snapshots should have distinct versions per node group.
+ */
+@AutoValue
+public abstract class V3Snapshot extends Snapshot {
```

Comment:
Why version prefixes in class names like here? To me that's what packages are for, i.e. `(...).v3.Snapshot`. Otherwise, we'll eventually remove v2 support but stay with awkward `V3Snapshot` and the like forever.